### ISSUE TITLE
Adjust commons

### ISSUE THREAD

**Issue by fubarhouse** [fubarhouse/ansible-role-jetbrains-upsource#10]
The virtual host system and the user identifier are common among many of my roles - it would seem sensible that I make those independent and give them an audit.

**Comment by fubarhouse** [comment]
So I've been able to identify specific tasks or functions into a commons role.

* assign fubarhouse_user if it hasn't been assigned.
* install a specified webserver via fubarhouse_webserver which uses geerlingguy roles for maximum compatibility - he's also quite experienced with these things.
* ensure virtual host folders are ready for use.
* on RedHat systems running apache, I've got a task to install proxy modules as this isn't covered by the geerlingguy.apache role - and I kinda need this for a lot of stuff on various ports.
* the standard will be updated to reflect the introduction of this new role.
* default apache2 modules for Ubuntu/Debian systems are specified.
* webservers are automatically installed as configured and are pulled in via the meta as a dependency, and they're only installed when fubarhouse_webserver is defined and configured as either 'apache2' or 'nginx'.

Any other common features which affect essentially all my roles will gradually have this extra dependency introduced in order to remove the common functionality from each role.

It's also important to note that I've gone back and retested the functionality of each vhost on each supported webhost. The apache one has been... buggy, for some time. This will completely resolve the problem.

Another thing I'm considering is to consolidate virtualhosts into the commons role via an array...
